### ISSUE TITLE
Character arrays for MAVLink messages must be exact length

### ISSUE THREAD

**Issue by Susurrus** [mavlink/mavlink#168]
The `Common:STATUSTEXT` message allows for 50 characters of characters. This message seems to imply that it's used for ASCII data. It's usage, however, can lead to some bugs when using the `mavlink_msg_statustext_pack()` function.

First off, `mavlink_msg_statustext_pack()` requires it's character array input to be 50 characters long. This means you can't do simple calls like:
```
mavlink_msg_statustext_pack(mavlink_system.sysid, mavlink_system.compid, &msg, MAV_SEVERITY_ERROR, "This is my message!");
```
This is because the string constant I put in there isn't 50 characters long. This would be a very subtle bug for people. Since the function prototype only calls for a `char *`, the need for it to be a full 50 character array isn't apparent.

For this reason I suggest adding removing the `char[]` from MAVLink to be replaced by both `string` and `int8_t/uint8_t`, where only `string` is currently missing. This new `string[LENGTH]` type would always be at least 1 char long and ALWAYS be null-terminated to allow for basic string processing functions to always be usable with it.

So now messages that want to send arbitrary data should use `uint8_t`, while those expecting strings should use the `string[LENGTH]` datatype. I would suggest that the `LENGTH` value follow from what `strlen()` would return, so it's the number of characters the string would hold, which is 1 less than the array needed to hold them. This makes this datatype consistent with `int8_t[LENGTH]`, where LENGTH indicates how many int8's can be held in that array.

An additional reason for my this change is necessary is because the `STATUSTEXT.text` field has the following descriptor:
>Status text message, without null termination character

What does that even mean? So if I use `strcpy()` or any of the standard string manipulation functions, it's actually violating the spec since there's a trailing '\0'? And if we don't terminate with a NULL character, how do we know where the end of the data is? We need to know somehow. This comment is exceedingly difficult to decipher.

**Comment by Susurrus** [comment]
Closing as the user can just implement this as needed, though I do think this would be nice for the C MAVLink generator to automatically include for any functions that deal with char[] arrays.

**Comment by bb4242** [comment]
This bug is old now, but I ran into this problem recently and was surprised to realize that mavlink was doing a buffer over-read when I passed in a string constant that wasn't exactly the length specified in the message definition.  This seems like dangerous behavior, as it could result in a segfault.  Just curious if there has been any further discussion of modifying the C mavlink generator to more gracefully handle char[] arrays?
